src/components/planning/DeleteEventsModal.vue

```diff
@@ -0,0 +1,101 @@
+<template>
+  <ni-modal v-model="value" @hide="hide">
+    <template slot="title"> Suppression d'interventions sur une période </template>
+    <ni-select in-modal caption="Bénéficiaire" v-model="deletedEvents.customer" :options="customersOptions"
+      required-field @blur="$v.deletedEvents.customer.$touch"
+      :error="$v.deletedEvents.customer.$error" />
+    <ni-option-group :displayCaption="false" v-model="deletedEvents.inRange" type="radio"
+      :options="deletetionOptions" inline />
+    <template v-if="deletedEvents.inRange">
+      <ni-date-input caption="Date de début" v-model="deletedEvents.startDate" type="date" required-field
+        in-modal @blur="$v.deletedEvents.startDate.$touch" :error="$v.deletedEvents.startDate.$error"/>
+      <ni-date-input caption="Date de fin" v-model="deletedEvents.endDate" type="date" required-field in-modal
+        :min="deletedEvents.startDate" @blur="$v.deletedEvents.endDate.$touch" :error="$v.deletedEvents.endDate.$error"/>
+    </template>
+    <template v-else>
+      <ni-date-input caption="Date d'arrêt des interventions" v-model="deletedEvents.startDate" type="date"
+        required-field in-modal/>
+    </template>
+    <template slot="footer">
+      <q-btn class="modal-btn full-width" color="primary" no-caps label="Supprimer les interventions" @click="deleteEvents"
+      icon-right="clear" />
+    </template>
+  </ni-modal>
+</template>
+<script>
+import { planningModalMixin } from '../../mixins/planningModalMixin';
+import Modal from '../Modal';
+import Select from '../form/Select';
+import OptionGroup from '../form/OptionGroup';
+import { NotifyNegative, NotifyPositive } from '../popup/notify';
+import { required, requiredIf } from 'vuelidate/lib/validators';
+import { validationMixin } from '../../mixins/validationMixin.js';
+
+export default {
+  name: 'DeleteEventsModal',
+  mixins: [planningModalMixin, validationMixin],
+  components: {
+    'ni-modal': Modal,
+    'ni-select': Select,
+    'ni-option-group': OptionGroup,
+  },
+  props: {
+    value: { type: Boolean, default: false },
+    customers: { type: Array, default: () => [] },
+  },
+  validations () {
+    return {
+      deletedEvents: {
+        customer: { required },
+        startDate: { required },
+        endDate: { required: requiredIf((item) => item.inRange) },
+      },
+    }
+  },
+  data () {
+    return {
+      deletedEvents: { inRange: true },
+      deletetionOptions: [
+        { label: 'Entre deux dates', value: true },
+        { label: 'A partir d\'une date', value: false },
+      ],
+    };
+  },
+  methods: {
+    hide () {
+      this.deletedEvents = { inRange: true };
+      this.$v.deletedEvents.$reset();
+      this.$emit('hide');
+    },
+    async validateEventsDeletion () {
+      this.$q.dialog({
+        title: 'Confirmation',
+        message: 'Es-tu sûr de vouloir supprimer ces évènements ?',
+        ok: 'OK',
+        cancel: 'Annuler',
+      }).onOk(this.deleteEvents())
```

Comment:
tu es sure que ca fonctionne comme ca ? 
j'aurai dit `onOk(this.deleteEvents)` ou `onOk(() => this.deleteEvents())

Reply:
et quand je supprime des evenements facturés, ca ne les supprime pas mais j'ai quand meme un message vert